src/snowflake/snowpark/dataframe.py

```diff
@@ -916,9 +916,17 @@ def to_pandas(
             2. If you use :func:`Session.sql` with this method, the input query of
             :func:`Session.sql` can only be a SELECT statement.
         """
-
         if _emit_ast:
-            raise NotImplementedError("TODO SNOW-1672576: Support to_pandas.")
+            stmt = self._session._ast_batch.assign()
+            ast = with_src_position(stmt.expr.sp_dataframe_to_pandas, stmt)
+            debug_check_missing_ast(self._ast_id, self)
+            if statement_params is not None:
+                for k, v in statement_params.items():
+                    t = ast.statement_params.add()
+                    t._1 = k
+                    t._2 = v
+            ast.block = block
+            self._set_ast_ref(ast.df)
 
         with open_telemetry_context_manager(self.to_pandas, self):
             result = self._session._conn.execute(
```

Comment:
the kwargs here prob. should contain. `dataframeAst`

Reply:
done